src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -497,7 +504,7 @@ def dataloader(self) -> Union[TRAIN_DATALOADERS, EVAL_DATALOADERS]:
         from pytorch_lightning import LightningDataModule, LightningModule  # prevent cyclic import
 
         if not self.name:
-            return self.instance
+            return self.instance  # type: ignore[return-value]
```

Comment:
do you know why we need to ignore here?

Reply:
https://github.com/Lightning-AI/lightning/pull/13806#discussion_r935491977